rust/rsc/src/bin/rsc/read_job.rs

```diff
@@ -34,26 +35,51 @@ async fn record_miss(hash: String, conn: Arc<DatabaseConnection>) {
 }
 
 #[tracing::instrument(skip(db, stores))]
-async fn resolve_blob(
-    id: Uuid,
-    db: &DatabaseTransaction,
+async fn resolve_blobs<T: ConnectionTrait>(
+    ids: &[Uuid],
+    db: &T,
     stores: &HashMap<Uuid, Arc<dyn blob::DebugBlobStore + Sync + Send>>,
-) -> Result<ResolvedBlob, String> {
-    let Ok(Some(blob)) = entity::prelude::Blob::find_by_id(id).one(db).await else {
-        return Err(format!("Unable to find blob {} by id", id));
-    };
+) -> Result<HashMap<Uuid, ResolvedBlob>, String> {
+    // Fetch all blobs in a single query
+    let blobs = Blob::find()
+        .filter(entity::blob::Column::Id.is_in(ids.to_vec()))
+        .all(db)
+        .await
+        .map_err(|e| format!("Failed to query blobs: {}", e))?;
 
-    let Some(store) = stores.get(&blob.store_id) else {
-        return Err(format!(
-            "Unable to find backing store {} for blob {}",
-            blob.store_id, id
-        ));
-    };
+    // Build a map of blob_id -> blob model for quick lookup
+    let blob_map: HashMap<Uuid, entity::blob::Model> = blobs.into_iter().map(|b| (b.id, b)).collect();
+
+    // Ensure we have all requested blobs
+    for &id in ids {
+        if !blob_map.contains_key(&id) {
+            return Err(format!("Unable to find blob {} by id", id));
```

Comment:
How does this error condition differ from the failed to query blobs check above?